src/crypto/algorithms/megolm.ts

```diff
@@ -277,37 +279,37 @@ class MegolmEncryption extends EncryptionAlgorithm {
         // Updates `session` to hold the final OutboundSessionInfo.
         //
         // returns a promise which resolves once the keyshare is successful.
-        const prepareSession = async (oldSession: OutboundSessionInfo) => {
-            session = oldSession;
-
+        const prepareSession = async (oldSession: OutboundSessionInfo | null) => {
             const sharedHistory = isRoomSharedHistory(room);
 
             // history visibility changed
-            if (session && sharedHistory !== session.sharedHistory) {
-                session = null;
+            if (oldSession != null && sharedHistory !== oldSession.sharedHistory) {
+                oldSession = null;
             }
 
             // need to make a brand new session?
-            if (session && session.needsRotation(this.sessionRotationPeriodMsgs,
+            if (oldSession != null && oldSession.needsRotation(this.sessionRotationPeriodMsgs,
```

Comment:
can we please use optional chaining? `oldSession?.needsRotation...`

Reply:
Whoops, did not catch that :+1: